src/builder/commands/ubuntu.rs

```diff
@@ -626,6 +645,46 @@ impl Distro {
             Ok(())
         }).map_err(|x| io::Error::new(io::ErrorKind::Other, x)).and_then(|x| x)
     }
+
+    #[cfg(feature="containers")]
+    fn copy_apt_archives_to_cache(&self, overlay_dir: &Path, cache_dir: &Path) -> io::Result<()> {
+        use rand::{thread_rng, Rng};
+        use rand::distributions::Alphanumeric;
+
+        let cache_dir = cache_dir.join("archives");
+        Dir::new(&cache_dir).create()?;
+
+        let archives_dir = overlay_dir.join("upper/archives");
+        if !archives_dir.exists() {
+            return Ok(());
+        }
+        ScanDir::files().read(&archives_dir, |iter| {
+            for (entry, name) in iter {
+                if !name.ends_with(".deb") { continue }
+                if !entry.file_type()?.is_file() { continue }
+                let src = entry.path();
+                let tmp_prefix: String = thread_rng()
+                    .sample_iter(&Alphanumeric)
+                    .take(6)
+                    .map(char::from)
+                    .collect();
+                let tmp = cache_dir.join(format!(".{}-{}.part", tmp_prefix, name));
```

Comment:
This should make into a function. We already have enough ad-hoc ways to create a tmp filename.